docs(deployment): sync design to online docs (#7256)
apache/apisix: docs/en/latest/architecture-design/deployment-role.md

---
title: Deployment Role
---

<!--
#
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
-->

> ## Concept

Reviewer: ```suggestion
##  Background
```

> Previously, the DP (data plane) and the CP (control plane) are not separate explicitly.

Reviewer: ```suggestion
Previously, the DP (Data Plane) and the CP (Control Plane) are not separate explicitly.

```

> Previously, the DP (data plane) and the CP (control plane) are not separate explicitly.
> Although we clearly distinguish the different responsibilities of DP and CP in the documentation, not everyone has correctly deployed APISIX in the production environment.

Reviewer: ```suggestion
Previously, the DP (Data Plane) and the CP (Control Plane) are not separate explicitly.

Although we clearly distinguish the different responsibilities of DP and CP in the documentation, not everyone has correctly deployed APISIX in the production environment.
```

> Therefore, we introduce new concepts called deployment modes/roles, to help users deploy APISIX easily and safely.
> APISIX under different deployment modes will act differently.

Reviewer: ```suggestion
Therefore, we introduce new concepts called deployment modes/roles, to help users deploy APISIX easily and safely.

APISIX under different deployment modes will act differently.
```

The table below shows the relationship among deployment modes and roles:

| Deployment Modes | Role                       | Description                                                                              |
|------------------|----------------------------|------------------------------------------------------------------------------------------|
| traditional      | traditional                | DP + CP are deployed together by default. People need to disable `enable_admin` manually |
| decoupled        | data_plane / control_plane | DP and CP are deployed independently.                                                    |
| standalone       | data_plane                 | Only DP, load the all configurations from local yaml file                                |

## Deployment Modes

### Traditional

![traditional](../../../assets/images/deployment-traditional.svg)

In the traditional deployment mode, one instance can be both DP & CP.

> There will be a `conf server` listens on Unix socket and acts as a proxy between APISIX and etcd.

Reviewer: ```suggestion

There will be a `conf server` listens on UNIX socket and acts as a proxy between APISIX and etcd.

```

> In the traditional deployment mode, one instance can be both DP & CP.
> There will be a `conf server` listens on Unix socket and acts as a proxy between APISIX and etcd.
> Both the DP part and CP part of the instance will connect to the `conf server` via HTTP protocol.

Reviewer: ```suggestion
In the traditional deployment mode, one instance can be both DP & CP.

There will be a `conf server` listens on UNIX socket and acts as a proxy between APISIX and etcd.

Both the DP part and CP part of the instance will connect to the `conf server` via HTTP protocol.
```

Here is the example of configuration:

```yaml title="conf/config.yaml"
deployment:
    role: traditional
    role_traditional:
        config_provider: etcd
    etcd:
       host:
           - http://xxxx
       prefix: /apisix
       timeout: 30
```

### Decoupled

![decoupled](../../../assets/images/deployment-cp_and_dp.svg)

The instance deployed as data_plane will:

> 1. Fetch configurations from control plane, the default port is 9280

Reviewer: ```suggestion
1. Fetch configurations from the CP, the default port is 9280
```

Reviewer: cause all the usages in the previous contents are DP and CP, recommended to change to CP here.

Although we clearly distinguish the different responsibilities of DP and CP in the documentation, not everyone has correctly deployed APISIX in the production environment.

Therefore, we introduce new concepts called deployment modes/roles, to help users deploy APISIX easily and safely.

APISIX under different deployment modes will act differently.

The table below shows the relationship among deployment modes and roles:

| Deployment Modes | Role                       | Description                                                                              |
|------------------|----------------------------|------------------------------------------------------------------------------------------|
| traditional      | traditional                | DP + CP are deployed together by default. People need to disable `enable_admin` manually |
| decoupled        | data_plane / control_plane | DP and CP are deployed independently.                                                    |
| standalone       | data_plane                 | Only DP, load the all configurations from local yaml file                                |

## Deployment Modes

### Traditional

> ![traditional](../../../assets/images/deployment-traditional.svg)

Reviewer 2: Please do not use a transparent background as the base color of the image.The picture will not be clear when enlarged.
![image](https://user-images.githubusercontent.com/97138894/173833272-e162471f-3719-43ae-8800-69db4ca0ef0e.png)

Author: I don't have the non-transparent version of the image. If there is not a simple way to add background, what about removing them?

Reviewer: I have edited those with white background, you can use that.

<img width="684" alt="11" src="https://user-images.githubusercontent.com/39793568/173970984-dee54556-cad0-4acf-ab01-250e32d4d773.png">

<img width="533" alt="22" src="https://user-images.githubusercontent.com/39793568/173971000-9bde954e-97d7-40a0-ab15-70e47d99e223.png">
